marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/artifacts/stepDefinition.sjs

```diff
@@ -53,17 +53,13 @@ function getArtifactNode(artifactName, artifactVersion) {
 function getDirectory(artifactName, artifact) {
     let doc = getArtifactNode(artifactName, null);
     let dir = "/step-definitions/";
-    let type;
     if(!doc && artifact && artifactName) {
         dir = dir + artifact.type.toLowerCase() + "/" + artifact.name +"/"
     }
     else if (doc) {
-        let mutableArtifact = doc.toObject();
-        if(mutableArtifact.name.startsWith("default-") || mutableArtifact.name == 'entity-services-mapping'){
-            dir = dir + mutableArtifact.toLowerCase() + "/" + "/marklogic/";
-        }
-        else {
-            dir = dir + mutableArtifact.type.toLowerCase() + "/" + mutableArtifact.name + "/";
+        let stepDefinition = doc.toObject();
+        if (stepDefinition.type && stepDefinition.name) {
+            dir = dir + stepDefinition.type.toLowerCase() + "/" + stepDefinition.name + "/";
```

Comment:
@rjrudin 
The ootb step definition's uri is /step-definitions/mapping/marklogic/default-mapping.step.json
The user defined step definition would be something like /step-definition/custom/myCustom/myCustom.step.json
Therefore, we may want to have this if-else.

Reply:
Had a conversation with @rjrudin , ootb step defs are loaded through v1/documents